homeassistant/components/plugwise/entity.py

```diff
@@ -67,7 +67,7 @@ def available(self) -> bool:
         """Return if entity is available."""
         return (
             self._dev_id in self.coordinator.data.devices
-            and ("available" not in self.device or self.device["available"])
+            and ("available" not in self.device or self.device["available"] is True)
```

Comment:
```suggestion
            and ("available" in self.device and self.device["available"] is True)
```

Reply:
That would require more logic to change, it would break the functionality to commit it as suggested

Reply:
This is not a good change, the reason for this specific construct is that the key "available" is not always present for all devices. It is not present for legacy-devices.

The logic is like this: when the key is not present the availability-status in HA must True, if it is present then the availability-status in HA is based on the state of `self.device["available"]`.

Reply:
```suggestion
            and ("available" not in self.device or self.device["available"])
```

Reply:
The `is True` part has been added to solve a mypy-error.

Reply:
> The is True part has been added to solve a mypy-error.

Please share, that seems highly unlikely as that is Python to check for a truthy value.
Unresolve this comment for that reason.

../Frenck

Reply:
@frenck while this would work and we tested it, it results in a typing validation `Incompatible return value type (got "bool | None", expected "bool")`. (And apologies for hitting the wrong commit suggestion button, UI's are awesome when they scroll automagically)

Reply:
Oooh right, as it is used as an expression. Ok makes sense 👍 

../Frenck